hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateHandler.java

```diff
@@ -380,70 +362,145 @@ private void updateExpiryFields(
-            validateTrue(originalToken.hasAdminKey(), TOKEN_IS_IMMUTABLE);
+        List<Key> lowPriorityKeys;
+        if (originalToken.hasAdminKey()) {
+            // if we update any of the low priority keys, we should allow either admin key
+            // or the respective low priority key to sign.
+            if (noOtherFieldThanLowPriorityKeyOrMetadataWillBeUpdated(op)) {
+                lowPriorityKeys = getAllRequiredLowPrioritySigners(originalToken, op);
+                Key requiredKey;
+                if (lowPriorityKeys.isEmpty()) {
+                    requiredKey = originalToken.adminKey();
+                } else {
+                    final Key lowPriorityKeyList = Key.newBuilder()
+                            .keyList(KeyList.newBuilder().keys(lowPriorityKeys).build())
+                            .build();
+                    final List<Key> keysRequired = List.of(lowPriorityKeyList, originalToken.adminKey());
+                    // with this we will require either admin key signature or all low priority keys signatures
+                    // (we can update several low priority keys with one TokenUpdate, so we will require all of these
+                    // keys
+                    // to sign)
+                    requiredKey = Key.newBuilder()
+                            .thresholdKey(ThresholdKey.newBuilder()
+                                    .keys(KeyList.newBuilder()
+                                            .keys(keysRequired)
+                                            .build())
+                                    .threshold(1)
+                                    .build())
+                            .build();
+                }
+
+                context.requireKey(requiredKey);
+            } else if (!isExpiryOnlyUpdateOp(op)) {
+                // if we update any other field, we need the current admin key to sign
+                // note: for expiry only op admin key is not required
+                context.requireKey(originalToken.adminKey());
+            }
+        } else {
+            lowPriorityKeys = getAllRequiredLowPrioritySigners(originalToken, op);
+            final Key lowPriorityKeyList = Key.newBuilder()
+                    .keyList(KeyList.newBuilder().keys(lowPriorityKeys).build())
+                    .build();
+            context.requireKey(lowPriorityKeyList);
         }
+    }
 
-        if (op.hasAdminKey()) {
-            final var newAdminKey = op.adminKey();
-            if (isKeyRemoval(newAdminKey)) {
-                builder.adminKey((Key) null);
-            } else {
-                builder.adminKey(newAdminKey);
+    private List<Key> getAllRequiredLowPrioritySigners(
+            @NonNull final Token originalToken, @NonNull final TokenUpdateTransactionBody op) {
+        List<Key> lowPriorityKeys = new ArrayList<>();
+        addLowPrioritySigner(lowPriorityKeys, originalToken.hasWipeKey(), originalToken.wipeKey(), op.hasWipeKey());
```

Comment:
hit : You don't need `originalToken.hasWipeKey()` parameter passed, instead check `originalToken.wipeKey() == null`

Reply:
Also for this you could use `isPresentInitially` and `isPresentInUpdate` in a for loop for all `TokenKeys` and add to `lowPriorityKeys` . That will be concise and more readable I think.